great_expectations/expectations/registry.py

```diff
@@ -312,8 +312,23 @@ def get_metric_provider(
 ) -> Tuple[MetricProvider, Callable]:
     try:
         metric_definition = _registered_metrics[metric_name]
+    except KeyError:
+        raise gx_exceptions.MetricProviderError(  # noqa: TRY003 # FIXME CoP
+            f"No metric named {metric_name} found."
+        )
+
+    try:
         return metric_definition["providers"][type(execution_engine).__name__]
     except KeyError:
+        # Search up class hierarchy for a match
+        for cls in type(execution_engine).mro()[1:]:
```

Comment:
I'm curious what lives at index 0 for `type(execution_engine).mro()`. Could use a comment since I know how some of these built-in methods are not always the most intuitive.

Reply:
It's the class name of the execution_engine itself so it's the same thing we already checked: `type(execution_engine).__name__`. I'll add a comment.